### ISSUE TITLE
RUF057 has false positives with starred arguments

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#21209]
### Summary

[`unnecessary-round` (RUF057)](https://docs.astral.sh/ruff/rules/unnecessary-round/) has false positives with starred arguments. [Example](https://play.ruff.rs/2269acbd-1a7c-4086-8e28-0d89d6c20907):
```console
$ cat >ruf057.py <<'# EOF'
print(round(125, **{"ndigits": -2}))
print(round(125, *[-2]))
# EOF

$ python ruf057.py
100
100

$ ruff --isolated check ruf057.py --select RUF057 --fix
Found 2 errors (2 fixed, 0 remaining).

$ cat ruf057.py
print((125))
print((125))

$ python ruf057.py
125
125
```

### Version

ruff 0.14.3 (8737a2d5f 2025-10-30)
